src/components/AddressView.tsx

```diff
@@ -1,29 +1,41 @@
-import { RootState } from '@store'
-import { AddressDisplay } from '@types'
 import * as React from 'react'
 import { connect } from 'react-redux'
+
+import { getActiveAccount } from '@selectors'
+import { getInactiveAccounts } from '@services/accounts'
+import { RootState } from '@store'
+import { AddressDisplay } from '@types'
+
 interface OwnProps {
   address: string
   showFull?: boolean
   addressDisplay: AddressDisplay
 }
 
-const mapStateToProps = (state: RootState, ownProps: OwnProps) => ({
-  addressInBook:
-    ownProps.addressDisplay === AddressDisplay.payee
-      ? state.contacts.contactList.find(contact => contact.address === ownProps.address)
-      : state.wallet.accounts.find(account => account.keypair.publicKey() === ownProps.address),
+const mapStateToProps = ({ contacts, wallet }: RootState) => ({
+  contactList: contacts.contactList,
+  accounts: wallet.accounts,
+  activeAccount: getActiveAccount(wallet),
 })
 
 type Props = OwnProps & ReturnType<typeof mapStateToProps>
 
 const AddressPresentation: React.SFC<Props> = props => {
+  const inactiveAccounts = getInactiveAccounts(props.accounts, props.activeAccount)
+
+  const addressInBook =
```

Comment:
why was this taken out of the mapStateToProps? it makes sense to me to do any processing of application state outside the presentation component, especially for such a small one

Reply:
I removed it from the props because I concatenated the first part of the ternary and thought it would be good to do it in one place